doc/classes/Parallax2D.xml

```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<class name="Parallax2D" inherits="Node2D" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
+	<brief_description>
+		A node used to create a parallax scrolling background.
+	</brief_description>
+	<description>
+		A [Parallax2D] is used to create a parallax effect. It can move at a different speed relative to the camera movement using [member scroll_scale]. This creates an illusion of depth in a 2D game. If manual scrolling is desired, the [Camera2D] position can be ignored with [member ignore_camera_scroll].
+		[b]Note:[/b] Any changes to this node's position made after it enters the scene will be ignored.
+	</description>
+	<tutorials>
+	</tutorials>
+	<members>
+		<member name="autoscroll" type="Vector2" setter="set_autoscroll" getter="get_autoscroll" default="Vector2(0, 0)">
+			Velocity at which the offset scrolls automatically, in pixels per second.
+		</member>
+		<member name="follow_viewport" type="bool" setter="set_follow_viewport" getter="get_follow_viewport" default="true">
+			If [code]true[/code], this [Parallax2D] is offset by the current camera's position. It may be desired to disable it if the [Parallax2D] is in a [CanvasLayer] separate from the current camera.
+		</member>
+		<member name="ignore_camera_scroll" type="bool" setter="set_ignore_camera_scroll" getter="is_ignore_camera_scroll" default="false">
+			If [code]true[/code], [Parallax2D]'s position is not affected by the position of the camera.
+		</member>
+		<member name="limit_begin" type="Vector2" setter="set_limit_begin" getter="get_limit_begin" default="Vector2(0, 0)">
+			Top-left limits for scrolling to begin. If the camera is outside of this limit, the [Parallax2D] stops scrolling. Must be lower than [member limit_end] minus the viewport size to work.
+		</member>
+		<member name="limit_end" type="Vector2" setter="set_limit_end" getter="get_limit_end" default="Vector2(0, 0)">
+			Bottom-right limits for scrolling to end. If the camera is outside of this limit, the [Parallax2D] will stop scrolling. Must be higher than [member limit_begin] and the viewport size combined to work.
+		</member>
+		<member name="offset" type="Vector2" setter="set_offset" getter="get_offset" default="Vector2(0, 0)">
+			The [Parallax2D]'s offset relative to its parent.
+		</member>
+		<member name="repeat_size" type="Vector2" setter="set_repeat_size" getter="get_repeat_size" default="Vector2(0, 0)">
+			The [Parallax2D]'s [Texture2D] repeating size. Useful for creating an infinite scrolling background. If an axis is set to [code]0[/code], the [Texture2D] will not be repeated.
+			If the texture's axis size multiplied by [member repeat_times] is smaller than the viewport axis, it will not repeat infinitely, as the [Parallax2D] only draws the specified number of instances of the texture at any given time and offsets it to create the effect.
```

Comment:
You're right. I think this isn't plain language at all, and it's hard to explain, honestly, which is why so many have trouble using it. The parallax effect in all of the parallax nodes (including this one) is achieved by repeating all the textures of its children offset by the specified amount. Then, when it scrolls and reaches the amount, it snaps back like a typewriter to seem like it's infinitely scrolling when it isn't, because the repeat effect is larger than the screen. [Kind of like this gif illustrates.](https://robotmafia.com/wp-content/uploads/2017/04/infinite_scroll.gif)

This leads to a few issues:
1. The user needs to say at what spot that cutoff should be, since it depends on the texture(s) used. i.e. using smaller textures and just setting it to the screen size is fine if you just want a cloud to repeat or something, but if you have a large background that should seamlessly transition at the cutoff point, it needs to be the exact size of the texture, because...
2. If the offset is smaller than the screen, you can visibly see it snapping back since doubling its size isn't large enough to cover the screen while it is scrolling.
3. With the old nodes, up until now, doubling was the limit, meaning that zooming out quickly undoes this infinite scroll effect since it's no longer large enough to cover the screen. I've added the ability to increase the number of repeats, allowing zooming out to be possible, but this adds another layer of complexity to explaining the effect.

So I'm definitely open to ideas for writing this out clearer, but it's unfortunately a complicated effect that's easy to use wrong if you don't know what it's doing.

Reply:
Your whole comment is very thorough can be kept in mind for later as inspiration. The verbose note could be removed in the meantime, as I feel like it's better to say nothing. It's not a deal breaker in my opinion, as just like `repeat_time`, the effect can be see in action relatively easily.

Reply:
I made a few more changes to the wording to be a little more clear to users.